students/jacob_daylong/Lesson 6/Mailroom_Pt4/mailroom_Pt4.py

```diff
@@ -0,0 +1,103 @@
+﻿import tempfile
+import sys
+
+full_name = ''
+table_header = ['Donor Name', 'Total Given', 'Num Gifts', 'Average Gift']
+donor_table = {}
+sorted_donors = {}
+dir = ()
+
+def menu():
+    dict_menu = {'\n1.': 'Send a Thank You', '2.' : 'Create a Report', '3.' : 'Send Thank You - All Donors', '4.' : 'Quit'}
+    for x, y in dict_menu.items():
+        print(x, y)
+
+def thankyou_note(fullname, donor_amt):
+    note = (f'\nDear {fullname}, \nThank you for your donation of ' 
+          f'${donor_amt:.2f}. \nSincerely, Jake\n')
+    return note
+
+def thankyou_notes(entry):
+    notes = (f'\nDear {entry}, \nThank you for your donation of ' 
+          f'${sum(donor_table.get(entry)):.2f}. \nSincerely, Jake\n')
+    return notes
+
+def donor_name_collection():
+     full_name = input("\nPlease enter donor's Full Name: ")
+     while full_name == 'List':
+        for entry in donor_table:
+            print(entry)
+        full_name = input("\nPlease enter donor's Full Name: ")
+     try:
+        user_input = input("Please enter donation amount:")
+        if user_input.isnumeric:
+            donor_amt = float(user_input)
+        else:
+            raise ValueError
+     except (ValueError, UnboundLocalError):
+        print("Please Enter a valid number")
+        donor_amt = float(input("Please enter donation amount:"))
+
+     send_thankyou(full_name, donor_amt)
+
+
+def send_thankyou(full_name, donor_amt):
+        if full_name not in donor_table:
+                donor_table[full_name] = [donor_amt]
+        else:
+                donor_table[full_name] += [donor_amt]
+    
+
+        print(thankyou_note(full_name, donor_amt))
+
+def thankyou_print():
+    for entry in donor_table:
+        dir = tempfile.gettempdir() + "/"
+        filename = entry + '.txt'
+        f = open(dir + filename, 'w')
+        f.write(thankyou_notes(entry))
+        f.close
```

Comment:
prefer to use `with` statement when working with files -- it will also make sure to close the file for you.